src/main/java/com/revature/assignforce/messaging/listener/CurriculumListener.java

```diff
@@ -0,0 +1,53 @@
+package com.revature.assignforce.messaging.listener;
+
+import java.io.IOException;
+import java.util.List;
+
+import org.springframework.amqp.rabbit.annotation.Exchange;
+import org.springframework.amqp.rabbit.annotation.Queue;
+import org.springframework.amqp.rabbit.annotation.QueueBinding;
+import org.springframework.amqp.rabbit.annotation.RabbitListener;
+import org.springframework.amqp.support.AmqpHeaders;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.messaging.handler.annotation.Header;
+import org.springframework.stereotype.Component;
+
+import com.rabbitmq.client.Channel;
+import com.revature.assignforce.beans.Batch;
+import com.revature.assignforce.service.BatchService;
+
+@Component
+public class CurriculumListener {
+
+	private BatchService batchService;
+	
+	private final String batchQueue;
+	
+	@RabbitListener(bindings = @QueueBinding(
+			value = @Queue(value = "batch-queue", durable = "true"),
+			exchange = @Exchange(value = "assignforce", ignoreDeclarationExceptions = "true"),
+			key = "assignforce.curriculum.delete")
+	)
+	public void receiveMessage(final Integer curriculumId, Channel channel, 
+				@Header(AmqpHeaders.DELIVERY_TAG) long tag) {
+		try {
```

Comment:
We should modify the order here. The ack should be sent when we've finished processing the message - that way the queue knows that the task is completed and it no longer needs to keep that item in the queue. If we send the ack first, then there is no guarantee that the task is actually completed and messages could be lost due to errors. Think of it like committing a transaction - we don't commit until all the work is done.